File: app/src/main/java/com/thibaudperso/sonycamera/timelapse/model/TimelapseData.java
package com.thibaudperso.sonycamera.timelapse.model;

import java.io.Serializable;

public class TimelapseData implements Serializable {

    private long mStartTime;
    private IntervalometerSettings mSettings;
    private ApiRequestsList mApiRequestsList;

    private boolean mIsFinished;

    public TimelapseData() {
        mSettings = new IntervalometerSettings();
        mApiRequestsList = new ApiRequestsList();
        mIsFinished = false;
    }

    public void clear() {
        mSettings = new IntervalometerSettings();
        mApiRequestsList.clear();
        mStartTime = 0;
        mIsFinished = false;
    }

    public long getStartTime() {
        return mStartTime;
    }

    public void setStartTime(long startTime) {
        this.mStartTime = startTime;
    }

    public IntervalometerSettings getSettings() {
        return mSettings;
    }

    public void setSettings(IntervalometerSettings settings) {
        mSettings = settings;
    }

    public ApiRequestsList getApiRequestsList() {
        return mApiRequestsList;
    }

    /*
     * Should not be called if it's not unlimited mode
     */
    public void setFinished(boolean finished) {
        mIsFinished = finished;
    }

    public boolean isTimelapseIsFinished() {
        return mIsFinished || !mSettings.isUnlimitedMode()
                && mSettings.framesCount == mApiRequestsList.getRequestsSent();
    }
}

File: app/src/main/java/com/thibaudperso/sonycamera/timelapse/model/TimelapseData.kt
package com.thibaudperso.sonycamera.timelapse.model

import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class TimelapseData @Inject constructor() {

    var startTime: Long = 0L
    var settings: IntervalometerSettings = IntervalometerSettings()
    val apiRequestsList: ApiRequestsList = ApiRequestsList()

    /*
     * Should not be called if it's not unlimited mode
     */
    var finished: Boolean = false

    val isTimelapseIsFinished: Boolean
        get() = finished || !settings.isUnlimitedMode && settings.framesCount == apiRequestsList.requestsSent

    fun clear() {
        settings = IntervalometerSettings()
        apiRequestsList.clear()
        startTime = 0L
        finished = false
    }


}
